classes/class09/README.md

```diff
@@ -1 +1,129 @@
+ - System overview
+ - References
+- Overall description
+ - Product perspective
+   - System Interfaces
+   - User Interfaces
+   - Hardware interfaces
+   - Software interfaces
+   - Communication Interfaces
+   - Memory Constraints
+   - Operations
+   - Site Adaptation Requirements
+ - Product functions
+ - User characteristics
+ - Constraints, assumptions and dependencies
+- Specific requirements
+ - External interface requirements
+ - Functional requirements
+ - Performance requirements
+ - Design constraints
+   - Standards Compliance
+ - Logical database requirement
+ - Software System attributes
+   - Reliability
+   - Availability
+   - Security
+   - Maintainability
+   - Portability
+ - Other requirements
+
+There is no hard and fast rule about how long a spec should be, but including all of that is just plain unnecessary for anything we'll be making in this class.
+Instead, let's distill some of the key ideas from above that our specs should touch on.
+
+***Disclaimer:*** Your teachers are by no means experts in writing specifications.
+The suggestions offered below are just that.
+The correct format will vary by project, so try to find what works best for you.
+
+- **Overview:** The story of your application, justification for why it needs to exist, and a broad overview of what it will accomplish.
+- **Use Cases:** What are some specific use cases you envision for the application?
+Who will use it and for what?
+- **User Interactions:** How will users interact with the software?
+- **Requirements:** Specific features that the application should implement.
+- **API/Feature Descriptions:** Specific details about functionality.
+You might define routes and what you expect them to do, or font-end ui elements that should be present.
+- **Test Cases:** How you plan to validate both your code and potentially the use cases?
+
+
+###TDD
+We learned a little bit about Test-Driven-Development last class, but we didn't go in to too much depth about using it for planning your application. 
+If you really take your TDD seriously, you can treat your tests as a high-level specification for the functionality of your application. 
+If you write all of your tests first, not only are you always validating your application, but you also have a fill-in-the-blank roadmap for development. 
+How convenient!
+
+###Diagramming
+process example
+
+####Logic Flow
+
+
+####Interactions
+
+
+####Network Layout
+
+
+##Refactoring
+
+
+###Folder Structure
+```
+MyApp/
++--public/
+|  +--javascripts/
+|     +--main.js
+|  +--css/
+|     +--main.css
+|  +--images/
+|     +--logo.png
++--views/
+|  +--layouts/
+|    +--main.handlebars
+|  +--partials/
+|     +--list_item.handlebars
+|  +--index.handlebars
+|  +--profile.handlebars
+|  +--login.handlebars
++--models/
+|  +--users.js
+|  +--twotes.js
++--routes/
+|  +--users.js
+|  +--twotes.js
++--util/
++--app.js
++--package.json
+```
+
+##Exercise
+Form groups of 4-5 people and select one of the following websites.
```

Comment:
We might specify (or maybe just in class) that each group should have a different site.